src/doc/rustc/src/codegen-options/index.md

```diff
@@ -181,30 +262,47 @@ This option is deprecated and does nothing.
 
 This flag lets you control debug information:
 
-* `0`: no debug info at all
+* `0`: no debug info at all (default)
 * `1`: line tables only
 * `2`: full debug info
 
+Note: The [`-g` flag][option-g-debug] is an alias for `-C debuginfo=2`.
+
 ## opt-level
 
 This flag lets you control the optimization level.
 
-* `0`: no optimizations, also turn on `cfg(debug_assertions)`.
+* `0`: no optimizations, also turn on [`cfg(debug_assertions)`](#debug-assertions).
 * `1`: basic optimizations
 * `2`: some optimizations
 * `3`: all optimizations
 * `s`: optimize for binary size
 * `z`: optimize for binary size, but also turn off loop vectorization.
 
+Note: The [`-O` flag][option-o-optimize] is an alias for `-C opt-level=2`.
+
+The default is `0`.
+
 ## debug-assertions
 
-This flag lets you turn `cfg(debug_assertions)` on or off.
+This flag lets you turn `cfg(debug_assertions)` [conditional
+compilation](../../reference/conditional-compilation.md#debug_assertions) on
+or off. It takes one of the following values:
+
+* `y`, `yes`, `on`, or no value: Enable debug-assertions.
+* `n`, `no`, or `off`: Disable debug-assertions.
+
+If not specified, debug assertions are enabled only if the
+[opt-level](#opt-level) is 0.
 
 ## inline-threshold
 
-This option lets you set the threshold for inlining a function.
+This option lets you set the threshold for inlining a function. It takes a
+positive integer as a value. Inlining is based on a cost model, where a higher
+threshold will allow more inlining.
 
-The default is 225.
+The default depends on the [opt-level](#opt-level). Current values are between
+25 to 275.
```

Comment:
I'm often reluctant to go into details on how optimizations work.  The correct settings usually require experimentation, and heavily depend on the project. They also are very LLVM specific, and are complex.  However, I can see how this is annoyingly vague.  I added a table, it can be updated or removed in the future if it ever changes.

Reply:
Sounds good, this can be resolved then.